atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/ServiceReadOnlyTransaction.java

```diff
@@ -0,0 +1,19 @@
+package com.palantir.atlasdb.transaction.impl;
+
+import com.palantir.atlasdb.transaction.api.AutoDelegate_Transaction;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.processors.AutoDelegate;
+
+@AutoDelegate(typeToExtend = Transaction.class)
+class ServiceReadOnlyTransaction implements AutoDelegate_Transaction {
```

Comment:
Do we want to support `HARD_DELETE` and `AGGRESSIVE_HARD_DELETE` for the service? If not, we could throw on some methods, to have a smaller public API.